android/src/com/mapswithme/maps/widget/DotPager.java

```diff
@@ -0,0 +1,182 @@
+ * <p>
+ * <ul>
+ * <li>@see {@link ViewPager}</li>
+ * <li>@see {@link PagerAdapter}</li>
+ * <li>An indicator. It's a {@link ViewGroup} which will consist dots. If the indicator
+ * is not needed this dependency can be missed or <code>null</code></li>
+ * <li>@see {@link Context}</li>
+ * <li>A page listener is an observable mechanism for listening page changing. It can be missed or null</li>
+ * </ul>
+ */
+public class DotPager implements ViewPager.OnPageChangeListener
+{
+  @NonNull
+  private final ViewPager mPager;
+  @NonNull
+  private final PagerAdapter mAdapter;
+  @Nullable
+  private final ViewGroup mIndicator;
+  @NonNull
+  private final ImageView[] mDots;
+  @NonNull
+  private final Context mContext;
+  @Nullable
+  private final OnPageChangedListener mListener;
+
+  private DotPager(@NonNull Builder builder)
+  {
+    mContext = builder.mContext;
+    mPager = builder.mPager;
+    mAdapter = builder.mAdapter;
+    mIndicator = builder.mIndicatorContainer;
+    mListener = builder.mListener;
+    mDots = new ImageView[mAdapter.getCount()];
+  }
+
+  public void show()
+  {
+    configure();
+    updateIndicator();
+  }
+
+  private void configure()
+  {
+    configurePager();
+    configureIndicator();
+  }
+
+  private void configurePager()
+  {
+    mPager.setAdapter(mAdapter);
+    mPager.clearOnPageChangeListeners();
+    mPager.addOnPageChangeListener(this);
+  }
+
+  private void configureIndicator()
+  {
+    if (mIndicator == null)
+      return;
+
+    mIndicator.removeAllViews();
+
+    if (mAdapter.getCount() == 1)
+      return;
+
+    for (int i = 0; i < mDots.length; i++)
+    {
+      mDots[i] = new ImageView(mContext);
+      LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
+          ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+      layoutParams.setMargins(0, 0, UiUtils.dimen(mContext, R.dimen.margin_half), 0);
+      mIndicator.addView(mDots[i], i, layoutParams);
+    }
+  }
+
+  @Override
+  public void onPageSelected(int position)
+  {
+    if (mIndicator != null)
+      updateIndicator();
+
+    if (mListener != null)
+      mListener.onPageChanged(position);
+  }
+
+  private void updateIndicator()
+  {
+    if (mAdapter.getCount() == 1)
+      return;
+
+    int currentPage = mPager.getCurrentItem();
+    for (int i = 0; i < mAdapter.getCount(); i++)
+    {
+      boolean isCurPage = i == currentPage;
+      @DrawableRes int dotDrawable;
+      if (ThemeUtils.isNightTheme())
+        dotDrawable = isCurPage ? R.drawable.news_marker_active_night : R.drawable.news_marker_inactive_night;
+      else
+        dotDrawable = isCurPage ? R.drawable.news_marker_active : R.drawable.news_marker_inactive;
+      mDots[i].setImageResource(dotDrawable);
+    }
+  }
+
+  @Override
+  public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels)
+  {
+    //no op
+  }
+
+  @Override
+  public void onPageScrollStateChanged(int state)
```

Comment:
1. In this case I cannot comment out parameter because this method is declared in  an `ViewPager.OnPageChangeListener` interface, i.e. the signature cannot be changed.
2. The syntax `void method(int /* state */)`  will cause compilation error in Java. 
3. In general, I absolutely agree with you that unused arguments are bad and I prefer to delete it at all in that case.

If I misunderstood your standpoint please let me know.